src/luxrays/core/namedobjectvector.cpp

```diff
@@ -141,43 +141,43 @@ vector<NamedObject *> &NamedObjectVector::GetObjs() {
 
 void NamedObjectVector::DeleteObj(const string &name) {
 	const u_int index = GetIndex(name);
-	objs.erase(objs.begin() + index);
 
-	// I have to rebuild the indices from scratch
-	name2index.clear();
-	index2obj.clear();
+	if (objs.size() > 0)
+	{
+		NamedObject* removedObj = objs[index];
+		int lastIndex = objs.size() - 1;
+		if (lastIndex != index)
+		{
+			NamedObject* lastObj = objs[lastIndex];
+			int newIndex = index;
+			objs[index] = lastObj;
+
+			// redo links
+			name2index.left.erase(removedObj->GetName());
+			name2index.left.erase(lastObj->GetName());
+			name2index.insert(Name2IndexType::value_type(lastObj->GetName(), index));
+
+			index2obj.left.erase(lastIndex);
+			index2obj.left.erase(index);
+			index2obj.insert(Index2ObjType::value_type(index, lastObj));
 
-	for (u_int i = 0; i < objs.size(); ++i) {
-		NamedObject *obj = objs[i];
-
-		name2index.insert(Name2IndexType::value_type(obj->GetName(), i));
-		index2obj.insert(Index2ObjType::value_type(i, obj));	
-	}
-}
-
-void NamedObjectVector::DeleteObjs(const vector<string> &names) {
-	vector<u_int> removeList;
-	removeList.reserve(names.size());
-	for (const string &name : names) {
-		if (IsObjDefined(name)) {
-			const u_int index = GetIndex(name);
-			removeList.push_back(index);
+		}
+		else
+		{
+			// ultimo
+			last.left.erase(removedObj->GetName());
```

Comment:
This line has to be
`name2index.left.erase(removedObj->GetName());`

Reply:
acc . copy and paste. this is my Original code i're right

	if (objs.size() > 0)
	{
		NamedObject* removedObj = objs[index];
		int lastIndex = objs.size() - 1;
		if (lastIndex != index)
		{
			NamedObject* lastObj = objs[lastIndex];
			int newIndex = index;
			objs[index] = lastObj;

			// rifaccio collegamenti
			name2index.left.erase(removedObj->GetName());
			name2index.left.erase(lastObj->GetName());
			name2index.insert(Name2IndexType::value_type(lastObj->GetName(), index));

			index2obj.left.erase(lastIndex);
			index2obj.left.erase(index);
			index2obj.insert(Index2ObjType::value_type(index, lastObj));

		}
		else
		{
			// ultimo
			name2index.left.erase(removedObj->GetName());
			index2obj.left.erase(index);
		}
	}